File: app-csharp/CharlesNadejda/CharlesNadejda/Models/CommandeWebLigne.cs
namespace CharlesNadejda.Models
{
    /// <summary>
    /// Ligne d'une commande web — un produit commandé avec sa quantité et son prix.
    /// Le sous-total est une colonne GENERATED côté MySQL (pas calculé en C#).
    /// </summary>
    public class CommandeWebLigne
    {
        // ── Colonnes DB (table commandes_web_lignes) ──────────────

        public int     Id             { get; set; }
        /// <summary>FK vers commandes_web.</summary>
        public int     IdCommande     { get; set; }
        /// <summary>FK vers produits_web.</summary>
        public int     IdProduitWeb   { get; set; }
        public int     Quantite       { get; set; }
        public decimal PrixUnitaire   { get; set; }
        /// <summary>Colonne GENERATED en DB = Quantite * PrixUnitaire.</summary>
        public decimal SousTotal      { get; set; }

        // ── Propriétés de jointure (chargées par le DAL) ──────────

        /// <summary>Chargé par CommandeWebDAL via JOIN produits_web.nom_commercial.</summary>
        public string NomProduit { get; set; }
    }
}

File: site-laravel/app/Models/CommandeWebLigne.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

/**
 * Ligne de commande web — associe un produit a une commande avec quantite et prix.
 *
 * Le prix_unitaire est fige au moment de l'ajout au panier (snapshot).
 *
 * @property int   $id
 * @property int   $id_commande
 * @property int   $id_produit_web
 * @property int   $quantite
 * @property float $prix_unitaire   Prix fige au moment de l'ajout.
 *
 * @property-read CommandeWeb $commande
 * @property-read ProduitWeb  $produit
 */
class CommandeWebLigne extends Model
{
    protected $table = 'commandes_web_lignes';
    public $timestamps = false;

    protected $fillable = [
        'id_commande', 'id_produit_web', 'quantite', 'prix_unitaire',
    ];

    /** Commande parente a laquelle appartient cette ligne. */
    public function commande(): \Illuminate\Database\Eloquent\Relations\BelongsTo
    {
        return $this->belongsTo(CommandeWeb::class, 'id_commande');
    }

    /** Produit web reference par cette ligne. */
    public function produit(): \Illuminate\Database\Eloquent\Relations\BelongsTo
    {
        return $this->belongsTo(ProduitWeb::class, 'id_produit_web');
    }
}
